src/cli/serve/serve.js

```diff
@@ -125,6 +125,11 @@ module.exports = function (program) {
       process.exit(1); // eslint-disable-line no-process-exit
     }
 
+    process.on('SIGTERM', function () {
+         kbnServer.close();
+         process.exit(0);
```

Comment:
Please drop the `process.exit()` call here. After `kbnServer.close()` is called the process should gracefully stop.